cloudformation/security-hq.template.yaml

```diff
@@ -331,7 +331,20 @@ Resources:
         Value:
           Fn::FindInMap: [ Constants, App, Value ]
         PropagateAtLaunch: true
-
+  TLSCertArn:
```

Comment:
I am uncertain if having the certificate created as part of the CloudFormation is 100% desirable? This presumes the existence of the DNS validation lambda and that the string passed in the `TLSCertDomain` parameter will be one that the lambda will be able to validate.

If this certificate fails to validate, what happens to the stack? Does it remain in create/update status? OR will it eventually timeout?

Reply:
> If this certificate fails to validate, what happens to the stack? Does it remain in create/update status? OR will it eventually timeout?

ACM in CloudFormation is blocking; the stack will be in create/update whilst the certificate is not validated. IIRC a certificate request times out after 72 hours and will cause the stack to fail.

I'm uncertain if we would ever change the domain of out instance of Security HQ. That is, our estate is setup to be happy with this change.

It sounds like we'd like to keep this repository approachable for external parties? If so, I think I agree, this change isn't needed and we should create the domain validated certificate manually and just update this parameter in the stack.